test(serialize): improved benchmarking

zsilbi opened the pull request:
I created a small util to automatically download specific versions from GitHub to compare with each other and the current version.

```ts
import { bench } from "vitest";
import { getVersions } from "./fixtures/utils/versions";

// Options for v1
const hashOptions = { unorderedArrays: true, unorderedSets: true };
const versions = await getVersions(["v1.1.5", "v2.0.4", "v2.0.11"]);
for (const version of versions) { 
  bench(version.name, () => {
    version.serialize("hello world", hashOptions); 
  });
}
```

Added custom benchmarks.
Added the same benchmarks for bun using `mitata`.
Added `bench:bun` command to run the bun benchmark.

pi0 commented:
Tip: we can use `"ohash-v1": "npm:ohash@^1"` in `devDependencies` to have same effect.

zsilbi commented:
> Tip: we can use `"ohash-v1": "npm:ohash@^1"` in `devDependencies` to have same effect.

This allows me to use any branch name, commit hash or tag to benchmark agains current (and each other).
If I only wanted to bench against v1 that would have been an easier solution of course 👍🏻

zsilbi commented:
I decided to remove `vitest` benchmarks for `serialize` and use only `mitata` for both Bun and `node` for easier maintainabilty.
Added `jiti` as dev. dep. to use `mitata`.
Added options to use arguments to turn on specific benchmarks (`--presets` (default) / `--combined` / `--custom` / `--full`).
